lib/std/mem.zig

```diff
@@ -630,6 +631,140 @@ test "spanZ" {
     testing.expectEqual(@as(?[:0]u16, null), spanZ(@as(?[*:0]u16, null)));
 }
 
+/// Takes a pointer to an array, an array, a sentinel-terminated pointer,
+/// or a slice and scans for the first 0, returning the scanned slice.
+/// If the type is an array or slice that is not sentinel terminated
+/// and no sentinel is found, the full length of the array/slice is returned.
+pub fn scanZ(ptr: anytype) Span(@TypeOf(ptr)) {
+    if (@typeInfo(@TypeOf(ptr)) == .Optional) {
+        if (ptr) |non_null| {
+            return spanZ(non_null);
+        } else {
+            return null;
+        }
+    }
+    const Result = Span(@TypeOf(ptr));
+    const l = scanLenZ(ptr);
+    if (@typeInfo(Result).Pointer.sentinel) |s| {
```

Comment:
`scanLenZ` handles `.Array` but this only handles `.Pointer`.  Since `scanLenZ` is only used here, they should be made to handle the same set of types.  It doesn't seem like handling of `.Array` is needed.